### PR TITLE
Handling warnings for unknown mimetypes

### PR DESCRIPTION
**Pull request by joelwilliam2005**
This pull request is towards fixing the issue #12484

### CODE REVIEW

**Comment by joelwilliam2005** [comment]
Should this work? @lb- @allcaps

**Comment by lb-** [comment]
@joelwilliam2005  were you able to run `make pdf` locally also?

I could not, I get an error as it's not set up in `make` yet, if I run `make latexpdfja` I get an error.

```
The LaTeX files are in _build/latex.
Run 'make' in that directory to run these through (pdf)latex
(use `make latexpdf' here to do that automatically).
Running LaTeX files through platex and dvipdfmx...
make -C _build/latex all-pdf-ja
make[1]: Entering directory '/code/wagtail/docs/_build/latex'
make[1]: *** No rule to make target 'all-pdf-ja'.  Stop.
make[1]: Leaving directory '/code/wagtail/docs/_build/latex'
make: *** [Makefile:131: latexpdfja] Error 2
root@226c0f160608:/code/wagtail/docs# 
```

Might need some further guidance from @allcaps  or @laymonage here, or maybe the actual building of PDFs does work differently on the readthedocs server. It looks like the latexpdf build runs to completion mostly, it does not warn about those mimetime items.

I will be heading offline now though, happy to review again in the morning.

File: .readthedocs.yml [hunk 1]
```diff
@@ -9,5 +9,6 @@ python:
       path: .
       extra_requirements:
         - docs
+formats: all
```
Comment (laymonage) [inline .readthedocs.yml:12]: The branch needs to be rebased, this is already in `main`

File: docs/conf.py [hunk 2]
```diff
@@ -327,6 +327,12 @@
 # If true, do not generate a @detailmenu in the "Top" node's menu.
 # texinfo_no_detailmenu = False
 
+epub_exclude_files = [
+    "_static/favicon.ico",
+    "_static/wagtail_colors_tables.txt",
+    "_static/wagtail_icons_table.txt",
+]
```
Comment (laymonage) [inline docs/conf.py:334]: Have you tried building the epub file locally? What happens to these files in the output?

An alternative may be to suppress the warning altogether, see https://github.com/sphinx-doc/sphinx/commit/6d900c34f12db0d21c581c58a5dd38ab49c8ea35 and https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-suppress_warnings

**Review by laymonage — COMMENTED** [review]
Thanks for giving this a go, but I suspect there's more to this than just excluding those files.

File: docs/conf.py (hunk 2 above)
Comment (laymonage) [reply docs/conf.py:334]: OK, looks like this doesn't make a difference in the `.epub` output:
image

Looking at the warnings:

```
WARNING: unknown mimetype for robots.txt, ignoring
WARNING: unknown mimetype for _static/favicon.ico, ignoring
WARNING: unknown mimetype for _static/wagtail_colors_tables.txt, ignoring
WARNING: unknown mimetype for _static/wagtail_icons_table.txt, ignoring
```

Looks like they got ignored anyway. So I think we should just suppress the warning, so we don't have to keep updating this list every now and then.

**Review by laymonage — COMMENTED** [review]
Thanks for giving this a go. I went with a different approach of suppressing the warnings in #12506.

I can see the files are now available for the `latest` version:

image

Though the generated ePub file has some errors in it. I've opened #12507 to fix the issues.

**Pull request closed without merging**